Tests/DiligentToolsTest/src/RenderStatePackager/RenderStatePackagerTest.cpp

```diff
@@ -0,0 +1,171 @@
+    ASSERT_TRUE(pEnvironment->Initilize());
+
+    auto pArchiveFactory = pEnvironment->GetArchiveFactory();
+
+    auto pConverter = pEnvironment->GetDeviceObjectConverter();
+
+    std::vector<std::string> InputFilePaths{"RenderStatesLibrary.drsn"};
+    ASSERT_TRUE(pConverter->ParseFiles(InputFilePaths));
+
+    RefCntAutoPtr<IArchiver> pArchive;
+    pArchiveFactory->CreateArchiver(pEnvironment->GetSerializationDevice(), &pArchive);
+    ASSERT_TRUE(pConverter->Execute(pArchive));
+
+    RefCntAutoPtr<IDataBlob> pData;
+    ASSERT_TRUE(pArchive->SerializeToBlob(&pData));
+}
+
+TEST(Tools_RenderStatePackager, PackagerIncorrectShaderPathTest)
+{
+    ParsingEnvironmentCreateInfo EnvironmentCI{};
+    EnvironmentCI.DeviceBits     = GetDeviceBits();
+    EnvironmentCI.RenderStateDir = "RenderStates/RenderStatePackager";
+    EnvironmentCI.ShaderDir      = "";
+
+    auto pEnvironment = std::make_unique<ParsingEnvironment>(EnvironmentCI);
+    ASSERT_TRUE(pEnvironment->Initilize());
+
+    auto pArchiveFactory = pEnvironment->GetArchiveFactory();
+
+    auto pConverter = pEnvironment->GetDeviceObjectConverter();
+
+    std::vector<std::string> InputFilePaths{"RenderStatesLibrary.drsn"};
+    ASSERT_TRUE(pConverter->ParseFiles(InputFilePaths));
+
+    RefCntAutoPtr<IArchiver> pArchive;
+    pArchiveFactory->CreateArchiver(pEnvironment->GetSerializationDevice(), &pArchive);
+
+    ASSERT_FALSE(pConverter->Execute(pArchive));
+}
+
+TEST(Tools_RenderStatePackager, PackagerIncorrectRenderStatePath)
+{
+    ParsingEnvironmentCreateInfo EnvironmentCI{};
+    EnvironmentCI.DeviceBits     = GetDeviceBits();
+    EnvironmentCI.RenderStateDir = "";
+    EnvironmentCI.ShaderDir      = "Shaders";
+
+    auto pEnvironment = std::make_unique<ParsingEnvironment>(EnvironmentCI);
+    ASSERT_TRUE(pEnvironment->Initilize());
+
+    auto pConverter = pEnvironment->GetDeviceObjectConverter();
+
+    std::vector<std::string> InputFilePaths{"RenderStatesLibrary.drsn"};
+    ASSERT_FALSE(pConverter->ParseFiles(InputFilePaths));
+}
+
+TEST(Tools_RenderStatePackager, PackagerMissingObjectsTest)
+{
+    ParsingEnvironmentCreateInfo EnvironmentCI{};
+    EnvironmentCI.DeviceBits     = GetDeviceBits();
+    EnvironmentCI.RenderStateDir = "RenderStates/RenderStatePackager";
+    EnvironmentCI.ShaderDir      = "Shaders";
+
+    auto pEnvironment = std::make_unique<ParsingEnvironment>(EnvironmentCI);
+    ASSERT_TRUE(pEnvironment->Initilize());
+
+    auto pArchiveFactory = pEnvironment->GetArchiveFactory();
+
+    auto pConverter = pEnvironment->GetDeviceObjectConverter();
+
+    {
+        std::vector<std::string> InputFilePaths{"MissingShader.drsn"};
+        ASSERT_TRUE(pConverter->ParseFiles(InputFilePaths));
+
+        RefCntAutoPtr<IArchiver> pArchive;
+        pArchiveFactory->CreateArchiver(pEnvironment->GetSerializationDevice(), &pArchive);
+
+        ASSERT_FALSE(pConverter->Execute(pArchive));
+        pConverter->Flush();
```

Comment:
I suggest renaming Flush to Clear or Reset.
Flush typically means writing out some buffer that was written, but this method simply resets the converter state.